apis/object/v1beta1/conversion.go

```diff
@@ -0,0 +1,22 @@
+/*
+Copyright 2023 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
```

Comment:
I am wondering if we should bump to `v1alpha2` instead of `v1beta1` 🤔 
The reasoning behind this is that we should not promote the [DependsOn](https://github.com/crossplane-contrib/provider-kubernetes/blob/6bb6320772f4c5fdc2f3361cd81a34f3a9e60579/apis/object/v1alpha1/types.go#L56) API and rather deprecate it in favor of XP `Usage`'s once usages become ">beta".

Reply:
Thats a good point. I agree that if we have/will have a native Crossplane way of doing it we shouldnt promote this one to Beta.

 Ill check if everything works fine if we migrate to `v1alpha2` instead of `v1beta1`

Reply:
Seems like its working fine. Changed the bump from `v1beta1` to `v1alpha2`